frontend/src/views/ClassroomsView.vue

```diff
@@ -0,0 +1,16 @@
+<script setup>
+import { ClassroomForm, ClassroomList } from '@/components/Classroom';
+import { ref } from 'vue';
+
+const reloadKey = ref(0);
```

Comment:
this part seems confusing, I think it would be better to move data fetching into this parent component rather than using hacks like this to reload data in component

Reply:
That approach was one of the options, but out of 3 solutions I found this was the simplest and best suited in my opinion because of the way I wrote and structured everything. Ideally, I wanted to create common components for form, table, etc. so I can reuse them and easily pass data. In that case I would use this approach but being that I went another way because of deadline I went with this approach.

Reply:
Cute way of relying on Vue's reactivity logic. But I consider this a form of abuse 😂 . I do not know what the third approach was, but something similar to what you did here is the following:
- a flag on this parent component that gets switched on when the form gets used
- the ClassromList components watches that flag (an example of where a watcher is needed; a piece of data outside of the component change that needs to trigger a method)
- the ClassroomList component emits an event after it fetches that data that triggers the flag in the parent to be switched off

This approach is similar to the one used here but it is more in line with the way Vue intends itself to be used. No need to reload the entire component when the data changes. That kind of undermines the point of a SPA.

I would still advise against either of these and do what @bdeak4 mentioned. But if that would not be done at this stage, then the one I propose above would have to suffice.